Firestore/core/test/firebase/firestore/remote/datastore_test.mm

```diff
@@ -353,6 +353,37 @@ void ForceFinishAnyTypeOrder(
   EXPECT_NO_THROW(credentials.InvokeGetToken());
 }
 
+// Error classification
+
+TEST_F(DatastoreTest, IsPermanentError) {
+  EXPECT_FALSE(
```

Comment:
This is fine.

You could optionally create a helper to make this more concise; something like:

```
EXPECT_NOT_PERMANENT_ERROR(FirestoreErrorCode::Cancelled);  // [1]
EXPECT_PERMANENT_ERROR(FirestoreErrorCode::DataLoss);

// [1] - Or EXPECT_RETRYABLE_ERROR, or similar.
```

Similarly for the write errors below.